api/src/main/java/io/opencensus/stats/AttachmentValue.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2019, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.stats;
+
+import com.google.auto.value.AutoValue;
+import io.opencensus.common.Function;
+import io.opencensus.stats.AggregationData.DistributionData.Exemplar;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * The value of {@link Exemplar} attachment.
+ *
+ * @since 0.20
+ */
+public abstract class AttachmentValue {
+
+  /**
+   * Applies the given match function to the underlying data type.
+   *
+   * @since 0.20
+   */
+  public abstract <T> T match(
```

Comment:
I don't think the match pattern is a good option here. Just have a:
```java
public abstract String getValue();
```

We can do a bit of a hack in the exporter and have:

```java
if (value instanceof AttachmentValueSpanContext) {
  AttachmentValueSpanContext avsc = (AttachmentValueSpanContext) value;
  avsc.getSpanContext();
}
```

Reply:
Agree that `match` pattern is not very useful in this case since there's only one subclass in the API. Having a `getValue` instead makes sense to me.